File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  // type your code here
  let arr = new Array(0,1)
  for (let i=1; i < num; i++)
    arr.push(arr[i-1] + arr[i])
    return arr[num]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// create an array with the first two numbers in the sequence
// iterate beginning at 1 since we want the position, not the index
// add numbers to the array as per the fibonacci sequence (each number is the sum of the previous two)
// return what's at the [num] position
// And a written explanation of your solution

// the sequence requires the first two numbers in order to make the third, so have to manually define those
// we continually creating new values in the sequence by adding the previous two, until we reach the position input
// we then return the number at that position
// this will create a very large array with a large input, however

File: 01-week-1--starter-algorithms/02-day-3--fibonacci-series/ruby/fibonacci.rb
def fibonacci(num) 
  # type your code in here
  array = [0,1]
  num.times.with_index(2) { |s, i| array.push(array[i-1] + array[i-2])}
  array[num]
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# create array with the first two in the sequence
# create the sequence by beginning at the 3rd index in the array and adding the previous two numbers
# num times
# return the value of the array at the num position
# And a written explanation of your solution

# times allows us to do an operation x times, in this case num times. 
# we begin at the three position because that is the first instance where we can add the previous two numbers
# keep adding new fibonacci numbers to the array until we break out of the loop
# return that number
